File: src-CilentReconfig/ProgressWidget.cpp
#include "ProgressWidget.h"
#include "ui_ProgressWidget.h"

#include <QThread>

ProgressWidget::ProgressWidget(QWidget *parent)
    : QDialog(parent),
      ui(new Ui::FileSend)
{
    ui->setupUi(this);
    ui->lineEdit->setReadOnly(true);
    ui->progressBar->setMaximum(100);
    this->setModal(false);
    this->setAttribute(Qt::WidgetAttribute::WA_DeleteOnClose); // 设置在关闭时自动删除
}

ProgressWidget::~ProgressWidget()
{
    delete ui;
}

void ProgressWidget::setTips(const QString &downloadOrSend)
{
    ui->label_2->setText(downloadOrSend);
}

void ProgressWidget::updateProgress(int progress)
{
    ui->progressBar->setValue(progress);
}

void ProgressWidget::setFilePath(QString path)
{
    ui->lineEdit->setText(path);
}

File: src-Server/progressWidget.py
from PySide6.QtCore import Qt, Slot
from PySide6.QtWidgets import QDialog

from ui_progressWidget import Ui_Form

"""进度窗口"""


class progressWidget(QDialog):
    def __init__(self, parent=None):
        super().__init__(parent)
        self.ui = Ui_Form()
        self.ui.setupUi(self)
        self.ui.ProgressRing.setMaximum(100)
        self.setModal(False)
        self.setAttribute(Qt.WidgetAttribute.WA_DeleteOnClose)  # 设置在关闭时自动删除

    @Slot()
    def updateProgress(self, progress: int):
        self.ui.ProgressRing.setValue(progress)

    def setTips(self, tip: str):
        self.ui.tips_label.setText(tip)
